Nested generics compilation error

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 2c42574, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 2c42574e6a25b4c5ca8381a63a05a0d0b997436f
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 专业版 26100 64-bit
|Processor           |24 cpus, 64bit, little endian, AMD Ryzen 9 5900X 12-Core Processor  
|Memory              |49.28GB/63.91GB
|                    |
|V executable        |D:\v\v\v.exe
|V last modified time|2025-05-18 14:17:11
|                    |
|V home dir          |OK, value: D:\v\v
|VMODULES            |OK, value: C:\Users\Administrator\.vmodules
|VTMP                |OK, value: C:\Users\Administrator\AppData\Local\Temp\v_0
|Current working dir |OK, value: E:\2025\v\v_test
|                    |
|Git version         |git version 2.46.0.windows.1
|V git status        |weekly.2022.48-5439-g2c42574e-dirty
|.git/config present |true
|                    |
|cc version          |cc (Rev5, Built by MSYS2 project) 13.2.0
|gcc version         |gcc (Rev5, Built by MSYS2 project) 13.2.0
|clang version       |clang version 18.1.6
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |N/A
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  nested_gen.v && E:\2025\v\v_test\nested_gen.exe`
```v
module main

struct People[T] {
	raw T
}

struct Man {
	name string
}

fn main() {
	// p1 := People{Man{
	// 	name: 'Mike'
	// }}
	m1 := Man{
		name: 'Tom'
	}

	gen1(m1)
}

fn gen1[T](m T) {
	gen2[People[T]]()
}

fn gen2[T]() {
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/Administrator/AppData/Local/Temp/v_0/nested_gen.tmp.c:535: warning: WINVER redefined
cc: C:/Users/Administrator/AppData/Local/Temp/v_0/nested_gen.tmp.c:1436: error: constant expression expected
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

```c
1434 VV_LOCAL_SYMBOL void main__main(void);
1435 VV_LOCAL_SYMBOL void main__gen1_T_main__Man(main__Man m);
1436 VV_LOCAL_SYMBOL void main__gen2_T_main__People[main__Man](void);
```

**What did you expect to see?**
`Compiled successfully`


Compiles successfully if the following section is commented out.
```v
module main

struct People[T] {
	raw T
}

struct Man {
	name string
}

fn main() {
	p1 := People{Man{
	 	name: 'Mike'
	 }}
	m1 := Man{
		name: 'Tom'
	}

	gen1(m1)
}

fn gen1[T](m T) {
	gen2[People[T]]()
}

fn gen2[T]() {
}

```




> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -754,7 +754,10 @@ fn (mut g Gen) fn_decl_params(params []ast.Param, scope &ast.Scope, is_variadic 
 			typ = g.table.sym(typ).array_info().elem_type.set_flag(.variadic)
 		}
 		param_type_sym := g.table.sym(typ)
-		mut param_type_name := g.styp(typ).replace_each(c_fn_name_escape_seq)
+		mut param_type_name := g.styp(typ)
+		if param.typ.has_flag(.generic) {
+			param_type_name = param_type_name.replace_each(c_fn_name_escape_seq)
+		}
 		if param_type_sym.kind == .function && !typ.has_flag(.option) {
 			info := param_type_sym.info as ast.FnType
 			func := info.func
@@ -2013,7 +2016,9 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 					}
 				}
 				name = g.generic_fn_name(concrete_types, name)
-				name = name.replace_each(c_fn_name_escape_seq)
+				if concrete_types.len > 0 {
+					name = name.replace_each(c_fn_name_escape_seq)
+				}
 			}
 		}
 	}

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -561,6 +561,7 @@ fn (mut g Gen) c_fn_name(node &ast.FnDecl) string {
 
 	if node.generic_names.len > 0 {
 		name = g.generic_fn_name(g.cur_concrete_types, name)
+		name = name.replace_each(c_fn_name_escape_seq)
 	}
 
 	if g.pref.translated || g.file.is_translated || node.is_file_translated {
@@ -2012,6 +2013,7 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 					}
 				}
 				name = g.generic_fn_name(concrete_types, name)
+				name = name.replace_each(c_fn_name_escape_seq)
 			}
 		}
 	}

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -2016,9 +2016,7 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 					}
 				}
 				name = g.generic_fn_name(concrete_types, name)
-				if concrete_types.len > 0 {
-					name = name.replace_each(c_fn_name_escape_seq)
-				}
+				name = name.replace_each(c_fn_name_escape_seq)
 			}
 		}
 	}

--- a/vlib/v/tests/generics/generic_fn_struct_test.v
+++ b/vlib/v/tests/generics/generic_fn_struct_test.v
@@ -1,0 +1,23 @@
+module main
+
+struct People[T] {
+	raw T
+}
+
+struct Man {
+	name string
+}
+
+fn test_main() {
+	m1 := Man{
+		name: 'Tom'
+	}
+	gen1(m1)
+}
+
+fn gen1[T](m T) {
+	gen2[People[T]]()
+}
+
+fn gen2[T]() {
+}

